source/plugins/go/src/extension/extension.go

```diff
@@ -94,6 +111,17 @@ func getDataTypeToStreamIdMapping() (map[string]string, error) {
 			logger.Printf("Info::mdsd::datatype: %s, outputstreamId: %s", dataType, outputStreamID)
 			datatypeOutputStreamMap[dataType] = outputStreamID.(string)
 		}
+		
+		//fetch dataCollectionSettings from extensionSettings and set to datatypeOutputStreamMap
+		extensionSettings := extensionConfig.ExtensionSettings
+		dataCollectionSettings := extensionSettings["dataCollectionSettings"]
+		containerLogV2Flag, isContainerLogV2FlagExists := dataCollectionSettings["containerLogV2"]
+		if isContainerLogV2FlagExists && containerLogV2Flag.(bool) {
+			logger.Printf("containerLogV2Flag value in DCR settings: %s", containerLogV2Flag)
+			datatypeOutputStreamMap["containerLogV2Flag"] = "true"

```

Comment:
This is confusing...Are we setting the value of datatypeOutputStreamMap to true always if the value is present in DCR?